### ISSUE TITLE
Dependency missing: `dask`

### ISSUE THREAD

**Issue by jkikstra** [IAMconsortium/concordia#66]
```
{
	"name": "ModuleNotFoundError",
	"message": "No module named 'dask'",
	"stack": "---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Cell In[4], line 1
----> 1 from concordia import RegionMapping

File c:\\Users\\kikstra\\Documents\\GitHub\\emissions_harmonization_historical\\.venv\\Lib\\site-packages\\concordia\\__init__.py:8
      5 from importlib.metadata import version as _version
      7 from .report import add_hypothesis, add_plotly_header, add_sticky_toc, embed_image
----> 8 from .utils import (
      9     CondordiaMagics,
     10     RegionMapping,
     11     VariableDefinitions,
     12 )
     15 try:
     16     __version__ = _version(\"concordia\")

File c:\\Users\\kikstra\\Documents\\GitHub\\emissions_harmonization_historical\\.venv\\Lib\\site-packages\\concordia\\utils.py:10
      7 from pathlib import Path
      8 from typing import Self, TypeAlias
---> 10 import dask.distributed as dd
     11 import numpy as np
     12 import pandas as pd

ModuleNotFoundError: No module named 'dask'"
}
```

**Comment by coroa** [comment]
That sounds like you did not install the conda environment from [environment.yml](https://github.com/IAMconsortium/concordia/blob/main/environment.yml)

**Comment by jkikstra** [comment]
You're right. And you're also right that that's is the way that it is currently supported - according to the [readme](https://iopscience.iop.org/article/10.1088/1748-9326/ac43df/meta).

I was doing it using `poetry`, which I suppose looks at the [pyproject.toml](https://github.com/IAMconsortium/concordia/blob/main/pyproject.toml).

**Comment by coroa** [comment]
It does look at that, but it's more by design than mistake. pyproject and pip/poetry and consorts in general have a very bad track record of working with anything that involves geospatial analysis around the gdal and osgeo libraries. It is in my experience safer to always manage environments with conda/mamba for those analyses. If zeb or others say that is no longer the case, i would be open to including them in pyproject.toml, but a few years back this has caused hard to debug headaches.

**Comment by znichollscr** [comment]
Nah all good @coroa thanks. We'll switch to pixi

**Comment by jkikstra** [comment]
Okidokes, so then I suppose this issue can be closed.

**Comment by coroa** [comment]
@znichollscr Thanks. pixi looks very helpful.

**Comment by znicholls** [comment]
Glad it helps. Our default environment managers are now basically:

- anything needing conda dependencies, use pixi (an example repo doing the whole set is here if you're interested: https://github.com/climate-resource/input4mips_validation)
- everything else, use uv (e.g. here: https://github.com/climate-resource/bookshelf)

In both cases, we use pdm as the build backend because it has some features we like, like locked builds (this repo basically demonstrates it https://github.com/openscm/openscm-zenodo). I reckon that uv will have their own version of this pretty soon, at which point we might stop using pdm.
